app/api/v1/auth/user.py

```diff
@@ -28,41 +34,75 @@
 router = APIRouter()
 
 
+@router.post(
+    "/find",
+    response_model=UserQueryResponse,
+    dependencies=[ScopedTo([UserScope.admin])],
+    tags=[USER],
+)
+async def find_users(
+    request: Request,
+    skip: int = 0,
+    limit: int = 100,
+    data: UserQueryRequest = Body(...),
+) -> UserQueryResponse:
+    """
+    Find users.
+
+    Search the repository for users that match the filter criteria specified in the request body.
+    If no filter criteria is specified the API will return all users.
+    """
+    filter = write_json_object(data.filter) if data.filter else {}
+    users = filter_user_info(filter, skip, limit, request.app.state.settings)
+    return UserQueryResponse(users=users)
+
+
+scoped_to_any = ScopedTo(
+    [UserScope.admin, UserScope.auditor, UserScope.guardian, UserScope.voter]
+)
+
+
 @router.get(
     "/me",
     response_model=UserInfo,
+    dependencies=[
+        ScopedTo(
+            [UserScope.admin, UserScope.auditor, UserScope.guardian, UserScope.voter]
+        )
+    ],
     tags=[USER],
 )
 async def me(
-    request: Request,
-    scopedTo: ScopedTo = ScopedTo(
-        [UserScope.admin, UserScope.auditor, UserScope.guardian, UserScope.voter]
-    ),
+    request: Request, token_data: ScopedTo = Depends(scoped_to_any)
 ) -> UserInfo:
     """
     Get user info for the current logged in user.
     """
-    token_data = scopedTo(request)
 
     if token_data.username is None:
         raise HTTPException(
             status_code=status.HTTP_400_BAD_REQUEST, detail="User not specified"
         )
 
     current_user = get_user_info(token_data.username, request.app.state.settings)
+
     if current_user.disabled:
         raise HTTPException(
             status_code=status.HTTP_400_BAD_REQUEST, detail="Inactive user"
         )
     return current_user
 
 
-@router.post(
-    "/create",
+@router.put(
```

Comment:
FYI changed create user from a POST to `user/create` to a PUT to `user`.  Seemed more consistent and like a cleaner API, hope it doesn't mess anyone up.

Reply:
There is probably a debate here. `POST` for creating when you don't know ID does make sense. `PUT` is technically inserting. I don't think this one should shift.

Reply:
PUT is typically updating as well.  I usually follow POST for creating and PUT for updating.  But it's not like we have to follow that.

Reply:
My primary concern is consistency.  Right now the following endpoints use PUT for inserts:

- election
- key ceremony
- key ceremony guardian

And the following use POST for inserts:

- guardian